### PR TITLE
Bootstrap check to see if the configured temporary directory is mounted with noexec

### PR DESCRIPTION
**Pull request by ph**
When you start logstash and the temporary directory is mounted with
`noexec` and you are using SSL this will make the netty library fails to
load the native library. This PR add a bootstrap check and a betters
message than missing a ciphers.

Fixes: #174

Notes for reviewer: Writing test for this isn't an easy task, I've tested it manually by creating a ramdisk with the `noexec` option and changing the config/logstash.yml directory to point to that.

### CODE REVIEW

File: lib/logstash/inputs/beats/stats_helper.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+      parts = path.split(::File::SEPARATOR)
+      number_of_iteration = parts.size
+
+      results = {}
+
+      number_of_iteration.times.collect do
+        new_path = ::File.join(*parts)
+
+        # we are interested all all the hierachy
```
Comment (ph) [inline lib/logstash/inputs/beats/stats_helper.rb]: double all.. :bummer.

File: lib/logstash/inputs/beats.rb [hunk 2]
```diff
@@ -175,6 +177,7 @@ def register
   def create_server
     server = org.logstash.beats.Server.new(@host, @port, @client_inactivity_timeout)
     if @ssl
+      check_no_exec_mount!
```
Comment (jordansissel) [inline lib/logstash/inputs/beats.rb]: Also, Netty has this https://netty.io/4.0/api/io/netty/handler/ssl/OpenSsl.html#isAvailable()

We can check OpenSsl.isAvailable(), or if we want the exception, call `OpenSsl.ensureAvailability()`. I have tried this in `tealess` and it fails when /tmp is noexec (or if the .so library fails to load in any way)

Comment (ph) [inline lib/logstash/inputs/beats.rb]: woa,  that make the code a lot simplier I wasn't aware of that method, I guess I've looked at the wrong place.

**Comment by jordansissel** [comment]
@ph did you still want to move this to use OpenSsl.isAvailable() ?

**Comment by ph** [comment]
Yes, it will be much better and simpler

**Comment by jordansissel** [comment]
I think we are agreed to move away from tcnative/ssl and toward Java ssl -- https://github.com/logstash-plugins/logstash-input-beats/issues/188

Closing this in favor of #188

**Pull request closed without merging**